### ISSUE TITLE
User feedback regarding performance

### ISSUE THREAD

**Issue by akaroml** [redhat-developer/vscode-java#857]
A user left some comments for Java Extension Pack and it's regarding language server performance.
<img width="1126" alt="Screen Shot 2019-03-25 at 8 30 55 AM" src="https://user-images.githubusercontent.com/16755094/54888549-e269d580-4ed8-11e9-8a18-05abfc691875.png">

There are two aspects to this problem:
1. The long lag for language server to load bigger projects (~8500 files in this case)
2. Progress update was not effective and the user gets the "stops responding" impression

While @yaohaizh is pushing for 1, @testforstephen could you please help revisit 2?

Here's the source of the comments:
https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack&ssr=false#review-details

**Comment by kelvinwop** [comment]
I feel like adding a short sleep every once in awhile would greatly help the "vscode has stopped responding" part. I mean the computation part is inevitable, but stopping once in awhile to check for user input would do wonders for responsiveness. Not sure how easy or hard that would be to implement though (or depending on design of code whether or not that would even help lol, just an idea).

**Comment by kelvinwop** [comment]
![image](https://user-images.githubusercontent.com/30770221/54896813-e9bac000-4e93-11e9-8088-fdfdc0ff6927.png)
some pics for effect

**Comment by yaohaizh** [comment]
@kelvinwop Could you attach your log information here. See how to get log information: https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting#enable-logging

**Comment by kelvinwop** [comment]
Aha, you're in luck! It just froze:
https://paste.ofcode.org/FzBfetrNzCFfPqyktjT3eK

**Comment by yaohaizh** [comment]
@kelvinwop Could you help disable other java extension(Debugger for Java,  Maven for Java, Java Dependency Viewer, Java Test Runner), leave only the Language Support for Java(TM) by redhat, and try it to see whether the perf issue hits?

**Comment by snjeza** [comment]
Related issues - https://github.com/eclipse/eclipse.jdt.ls/issues/824, https://github.com/Microsoft/vscode-java-pack/issues/81

**Comment by kelvinwop** [comment]
Will do,

Extra info: when I get this notification: 
![image](https://user-images.githubusercontent.com/30770221/54945812-afd5d200-4f04-11e9-872a-7c729462cf43.png)

there is no lag ever bug all the "find references/control click trace/etc" break beyond all hell

**Comment by akaroml** [comment]
> Related issues - [eclipse/eclipse.jdt.ls#824](https://github.com/eclipse/eclipse.jdt.ls/issues/824), [Microsoft/vscode-java-pack#81](https://github.com/Microsoft/vscode-java-pack/issues/81)

Here's more on Microsoft/vscode-java-pack#81. It turns out to be a vscode issue. We observed lots of such issues recently and got no luck in all the investigations. And after some time, other extension authors started to receive similar issues, and then we started to suspect it was a vscode issue. @eskibear has more info and will update that issue. It will be closed soon.

**Comment by kelvinwop** [comment]
Some more info: after disabling the other extensions, I got the Classpath is incomplete error on every file. After enabling it again, the lag went away and the Find By References broke. In any case, I've moved to android studio for the android sdk support. Thanks!

**Comment by kelvinwop** [comment]
Was working on a small proj (for a change) this is completely unrelated but coloring seems to be a bit broken for for loops:
![image](https://user-images.githubusercontent.com/30770221/55039446-23a1d880-4ff2-11e9-9570-b6415fcbe315.png)

**Comment by yaohaizh** [comment]
@kelvinwop So you open Android project in VSCode. If this is the case, VSCode Java has not support Android yet. See the related issue: https://github.com/redhat-developer/vscode-java/issues/396

**Comment by kelvinwop** [comment]
Yeah I realized that pretty quick, but funnily, vscode's file search/function tracing is better than intelliJ/android studio, despite having pages upon pages of errors. Anyways I finished most of the tracing work so I'm now moving between IDEs depending on what I'm doing

**Comment by apupier** [comment]
in 2019:

> It will be closed soon.

@akashshinde @Eskibear any update on this issue?
